app/integrations/weather_api.py

```diff
@@ -12,44 +18,52 @@
 load_dotenv()
 API_KEY = os.getenv("WEATHER_API_KEY")
 CITY = "Brisbane"
+FIFTEEN_MINS = 900
 
-# Current weather.
-current_query = (
-    f"https://api.weatherapi.com/v1/current.json?key={API_KEY}&q={CITY}&aqi=no"
-)
-
-forecast_query = f"https://api.weatherapi.com/v1/forecast.json?key={API_KEY}&q={CITY}&days=3&aqi=no&alerts=no"
+# Copy paste these into functions as you need them. Can delete once they've been used.
+# current_query = (
+#     f"https://api.weatherapi.com/v1/current.json?key={API_KEY}&q={CITY}&aqi=no"
+# )
+# forecast_query = f"https://api.weatherapi.com/v1/forecast.json?key={API_KEY}&q={CITY}&days=3&aqi=no&alerts=no"
 
 
 @weather_api_bp.route("/get_weather", methods=["GET"])
 def get_weather():
     now = int(time.time())
-    collected_at = db.fetch_collected_at()
 
-    if (now - int(collected_at)) > 300:
-        fetch_external_data(now)
+    last_updated_stmt = select(Weather.id).where(Weather.key == "last_updated_epoch")
+    last_updated = db.session.execute(last_updated_stmt).scalar_one_or_none()
 
-    db_data = db.fetch_all("current_weather")
+    if (last_updated is None) or (now > last_updated + FIFTEEN_MINS):
+        fetch_external_data()
 
-    print(db_data)
+    fetch_all_query = select(Weather)
+    db_data = db.session.execute(fetch_all_query).scalars().all()
 
-    return db_data
+    return jsonify({row.key: row.value for row in db_data})
```

Comment:
A simple test for this shouldn't be too hard. Worth writing at least one to make tests easier to write later.

The main complesity is to patch `requests.get`